wavelength-plugins/emoticons.js

```diff
@@ -105,33 +125,61 @@ exports.commands = {
 			}
 		},
 
-		'': 'view',
-		list: 'view',
-		view: function (target, room, user) {
+		view: "list",
+		list: function (target, room, user) {
 			if (!this.runBroadcast()) return;
-			let reply = `<b><u>Emoticons (${Object.keys(emoticons).length})</u></b><br />`;
-			for (let emote in emoticons) reply += `(${emote} <img src="${emoticons[emote]}" height="40" width="40">) `;
+
+			let size = 50;
+			let lobby = Rooms("lobby");
+			if (lobby && lobby.emoteSize) size = lobby.emoteSize;
+			if (room.emoteSize) size = room.emoteSize;
+
+			let reply = `<strong><u>Emoticons (${Object.keys(emoticons).length})</u></strong><br />`;
+			for (let emote in emoticons) reply += `(${emote} <img src="${emoticons[emote]}" height="${size}" width="${size}">)`;
 			this.sendReply(`|raw|<div class="infobox infobox-limited">${reply}</div>`);
 		},
 
 		ignore: function (target, room, user) {
-			if (WL.ignoreEmotes[user.userid]) return this.errorReply("You are already ignoring emoticons.");
+			if (WL.ignoreEmotes[user.userid]) return this.errorReply(`You are already ignoring emoticons.`);
 			WL.ignoreEmotes[user.userid] = true;
-			fs.writeFileSync('config/ignoreemotes.json', JSON.stringify(WL.ignoreEmotes));
+			FS(`config/ignoreemotes.json`).writeSync(JSON.stringify(WL.ignoreEmotes));
 			this.sendReply(`You are now ignoring emoticons.`);
 		},
 
 		unignore: function (target, room, user) {
-			if (!WL.ignoreEmotes[user.userid]) return this.errorReply("You aren't ignoring emoticons.");
+			if (!WL.ignoreEmotes[user.userid]) return this.errorReply(`You aren't ignoring emoticons.`);
 			delete WL.ignoreEmotes[user.userid];
-			fs.writeFileSync('config/ignoreemotes.json', JSON.stringify(WL.ignoreEmotes));
+			FS(`config/ignoreemotes.json`).writeSync(JSON.stringify(WL.ignoreEmotes));
 			this.sendReply(`You are no longer ignoring emoticons.`);
 		},
 
+		size: function (target, room, user) {
+			if (room.id === `lobby` && !this.can(`emote`) || room.id !== `lobby` && !this.can(`emote`, null, room)) return false;
+			if (!target) return this.sendReply(`Usage: /emoticons size [number]`);
+
+			let size = Math.round(Number(target));
+			if (isNaN(size)) return this.errorReply(`"${target}" is not a valid number.`);
+			if (size < 1) return this.errorReply(`Size may not be less than 1.`);
+			if (size > 200) return this.errorReply(`Size may not be more than 200.`);
```

Comment:
maximum should be 100 anything else is excessively large, also some of the emotes will become blurry because of their actual size being 40x40 or 50x50